### ISSUE TITLE
Question: Regarding time complexity of Oversamplers and "Noise Filters"

### ISSUE THREAD

**Issue by BrandonKMLee** [analyticalmindsltd/smote_variants#60]
For Scikit Learn some have created tools for demoing latency (model fitting) against error.
- https://scikit-learn.org/0.15/modules/computational_performance.html
- https://scikit-learn.org/stable/auto_examples/applications/plot_model_complexity_influence.html

The Scitime estimator is useful for some of the algorithms in Scikit-learn but not all
- https://www.freecodecamp.org/news/two-hours-later-and-still-running-how-to-keep-your-sklearn-fit-under-control-cc603dc1283b/
- https://github.com/scitime/scitime

It would be useful to measure the time complexity of oversamplers and see which ones are fast (or not) based on size of dataset and log-odds of majority proportions.

**Comment by gykovacs** [comment]
I agree, the time complexity of oversampling techniques is somewhat unexplored. Some runtime measurements are incorporated though. There was an extensive evaluation (shared in corresponding papers), and based on the average runtimes on 104 datasets a ranking of oversampling techniques is available. For example, if one is interested in the 10 quickest techniques overall, then can query them as

```python
import smote_variants as sv

# get 10 quickest oversamplers
oversamplers = sv.get_all_oversamplers(n_quickest=10)
```

Although this is not a true time complexity analysis, it can still be used to query computationally efficient techniques for further research or application purposes.

Nevertheless, a proper time complexity analysis by varying number of majority and minority samples, features, imbalance ratios class overlap, etc. would be very useful.

Regarding the noise filters, they are not intended to be primary or necessary steps for oversampling pipelines, but similar analysis on them could still be useful.
